### PR TITLE
Fix #312: Update SLUG_RE to accept backticks

### LINKED ISSUE #312: User page is 404 when username contains a backtick (`)
My IRC nickname is *NiKo`* (notice the backtick at the end) and it seems the system doesn't accept this char in the urls, so my page is 404 : https://www.standu.ps/user/NiKo%60/

Also and as a probable side effect, my entries are not linked to my personal profile page but to [the team's one](https://www.standu.ps/project/storage-team/) (though maybe that's a setting? idk).

Feels like some regex needs to be updated anyway. (Edit: #313)

### PR DESCRIPTION
**Pull request by n1k0**
Refs #312. Sorry I didn't test the patch, I've just edited this file right from github. Feel free to close or take over if that's not the right way to properly address the issue :)

### CODE REVIEW

**Comment by glasserc** [comment]
If this is meant to match IRC nicks, it seems like https://tools.ietf.org/html/rfc1459#section-2.3.1 says that nicks can include "special" characters including '-', '[', ']', '\', '`', '^', '{', and '}'.

**Comment by n1k0** [comment]
I couldn't manage including the backslash character as Django seems to intercept it early and rewrite it to a regular slash.

File: standup/status/migrations/0005_add_team_user_id.py [hunk 1]
```diff
@@ -21,11 +21,6 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        migrations.AddField(
-            model_name='teamuser',
-            name='id',
-            field=models.AutoField(serialize=False, auto_created=True, verbose_name='ID')
-        ),
```
Comment (n1k0) [inline standup/status/migrations/0005_add_team_user_id.py]: See #314

**Comment by n1k0** [comment]
Hmm this patch may just be plain wrong after all. After reading the codebase, it appears we should always rely on Django's `User` model `username` property to resolve user urls. So, a few thoughts:

- Maybe what needs to be updated first is the [standup_irc](https://github.com/mozilla/standup-irc) bot, which is generating erroneous urls;
- But as it consumes the API, maybe what should be updated as well is its views exposing user information, so the bot has access to the appropriate data to forge urls;
- The Auth0 integration seems to [automatically generate usernames](https://github.com/mozilla/standup/blob/ddfb37bd07e5833986fe2622ac59badc92d6df50/standup/auth0/utils.py#L21-L30), maybe a good idea would be to ask users to provide one they're happy with instead? So we would also have our unique slugified identifier to build proper urls.

I'm gonna move to other work I have to do, hope this investigation was somewhat helpful.

**Comment by willkg** [comment]
You're seeing issue #279. After we fix that, the problem you're hitting should go away. I'm not sure we should additionally adjust slugs, but we can create a new issue for that and discuss it there if you like.

For now, I'm going to close this out.

**Pull request closed without merging**